src/transformers/testing_utils.py

```diff
@@ -2497,7 +2500,47 @@ def wrapper(*args, **kwargs):
                     return test_func_ref(*args, **kwargs)
 
                 except Exception as err:
-                    print(f"Test failed with {err} at try {retry_count}/{max_attempts}.", file=sys.stderr)
+                    logger.error(f"Test failed with {err} at try {retry_count}/{max_attempts}.")
+                    if wait_before_retry is not None:
+                        time.sleep(wait_before_retry)
+                    retry_count += 1
+
+            return test_func_ref(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+def hub_retry(max_attempts: int = 5, wait_before_retry: Optional[float] = 2):
+    """
+    To decorate tests that download from the Hub. They can fail due to a
+    variety of network issues such as timeouts, connection resets, etc.
+
+    Args:
+        max_attempts (`int`, *optional*, defaults to 5):
+            The maximum number of attempts to retry the flaky test.
+        wait_before_retry (`float`, *optional*, defaults to 2):
+            If provided, will wait that number of seconds before retrying the test.
+    """
+
+    def decorator(test_func_ref):
+        @functools.wraps(test_func_ref)
+        def wrapper(*args, **kwargs):
+            retry_count = 1
+
+            while retry_count <= max_attempts:
+                try:
+                    return test_func_ref(*args, **kwargs)
+                # We catch all exceptions related to network issues from requests
+                except (
+                    requests.exceptions.ConnectionError,
+                    requests.exceptions.Timeout,
+                    requests.exceptions.ReadTimeout,
+                    requests.exceptions.HTTPError,
+                    requests.exceptions.RequestException,
+                ) as err:
+                    logger.error(f"Test failed with {err} at try {retry_count}/{max_attempts}.")
```

Comment:
It might be better to write as "Failed to connect to the specified Hub repository ..."

(so the message is more clear, and different from `is_flaky`)

Reply:
after looking at the message I got from slack yesterday

https://buildkite.com/vllm/ci/builds/13929#01952a60-e9cd-4a0b-8582-e56b3ebf1c18/195-1735

i am somehow unsure if the list in `except` covers other failing case (like the above link), which is 

> OSError: We couldn't connect to 'https://huggingface.co' to load this file, couldn't find it in the cached files and it looks like ArthurZ/Ilama-3.2-1B is not the path to a directory containing a file named modeling_ilama.py.

but that is caused first by `requests.exceptions.ReadTimeout`.

cc @Wauplin for this part, maybe you have some ideas regarding such situations.